examples/seq2seq/finetune.py

```diff
@@ -102,14 +103,13 @@ def __init__(self, hparams, **kwargs):
 
         self.hparams.git_sha = get_git_info()["repo_sha"]
         self.num_workers = hparams.num_workers
-        self.decoder_start_token_id = None
+        self.decoder_start_token_id = None  # default to config
         if self.model.config.decoder_start_token_id is None and isinstance(self.tokenizer, MBartTokenizer):
             self.decoder_start_token_id = self.tokenizer.lang_code_to_id[hparams.tgt_lang]
             self.model.config.decoder_start_token_id = self.decoder_start_token_id
-        if isinstance(self.tokenizer, MBartTokenizer) or isinstance(self.tokenizer, MarianTokenizer):
-            self.dataset_class = TranslationDataset
-        else:
-            self.dataset_class = Seq2SeqDataset
+        self.dataset_class = (
```

Comment:
important change

Reply:
If every seq2seq model is going to have the `prepare_seq2seq_batch` then we might not need `LegacyDataset`. Would it be a good idea to make make `prepare_seq2seq_batch` batch mandatory for all seq2seq models and not maintain two datasets ?

Reply:
Maybe eventually. @patrickvonplaten is experimenting with composing GLUE models into seq2seq models with `EncoderDecoderModel`, (e.g. `Roberta2Roberta`), so I think that we keep the Legacy dataset until we can guarantee that most tokenizers have `prepare_seq2seq_batch` methods, which might be never.